test/functional/test_site/index.md

```diff
@@ -43,7 +43,7 @@ head: myCustomHead.md, myCustomHead2.md
 {{ page_variable_referencing_page_variable }}
 
 # Global Variable overriding Page Variable
-<variable name="page_global_variable_overriding_page_variable">Page Variable overridden by Global Variable</variable>
+<variable name="page_global_variable_overriding_page_variable">**TEST FAIL**: Page Variable overridden by Global Variable</variable>
```

Comment:
Instead of `**TEST FAIL**`, maybe use `**Incorrect input**` or `**Should not appear**`?

Also, I was wondering whether there will be any value in converting some of these tests to unit tests, which might allow a cleaner implementation? Agree that generally it is difficult to read the .html output, although we can also consider breaking up `index.html` further because it is getting rather huge and messy. :P

Reply:
> ```markdown
> # Global Variable overriding Page Variable
> ```

Also since headings usually generate quite a long html code (because it also generates the anchor ID and the anchor icon itself), perhaps we can minimize the use of headings (only tests that are testing the features of headings should use it).

Reply:
Hmm sure I'll keep that in mind, I agree it might be helpful to do something about index before it gets too large

How about we convert them to `h4`s so they don't get indexed?

I'll use **Should not appear** as it is more descriptive

Reply:
`h4` still generates an unnecessary ID though.

```html
<h4 id="global-variable-overriding-page-variable">Global Variable overriding Page Variable</h4>
```

How about just using bold?

```markdown
**Global Variable overriding Page Variable**
```

Reply:
Sure - I can handle that in a separate PR.

In the meantime is this one okay?